arangod/GeneralServer/AsyncJobManager.cpp

```diff
@@ -125,16 +141,14 @@ void AsyncJobManager::deleteJobs() {
   auto it = _jobs.begin();
 
   while (it != _jobs.end()) {
-    GeneralResponse* response = (*it).second._response;
-
-    if (response != nullptr) {
-      delete response;
+    if (::authorized(it->second)) {
```

Comment:
Is `it` increased in the `else` case too?
I think this is missing, so we will have an endless loop hee.